runners/spark/src/main/java/org/apache/beam/runners/spark/SparkRunner.java

```diff
@@ -147,12 +152,22 @@ public SparkPipelineResult run(final Pipeline pipeline) {
 
     MetricsEnvironment.setMetricsSupported(true);
 
+    // visit the pipeline to determine the translation mode
     detectTranslationMode(pipeline);
 
+    // create the cache candidates map
+    Map<PCollection, Long> cacheCandidates = new HashMap<>();
+
     if (mOptions.isStreaming()) {
       CheckpointDir checkpointDir = new CheckpointDir(mOptions.getCheckpointDir());
       final SparkRunnerStreamingContextFactory contextFactory =
           new SparkRunnerStreamingContextFactory(pipeline, mOptions, checkpointDir);
+      // update cache candidates
+      SparkPipelineTranslator translator = new StreamingTransformTranslator.Translator(
```

Comment:
`SparkPipelineTranslator translator` could be a member of SparkRunner since it is instantiated both for streaming and batch executions.